frontend/src/js/util/LruCache.ts

```diff
@@ -85,6 +85,16 @@ export class LruCache<K extends string | number, V> {
   };
 
   replace = (k: K, v: V): void => {
-    this.addToCache(k, v);
+    this.entries[k] = v;
```

Comment:
I think we should check before modifying the cache that the entry is indeed already in there. Otherwise I think `bumpRecency` might blow up because it assumes `k` is in the recency queue. (And if it doesn't, we'd end up with the recency queue and the cache out of sync again).

Alternatively, you could continue using `addToCache` but check whether `k` was already in the cache before adding it to the recency queue.

Reply:
Great point @joelochlann  I put a condition within the replace itself to check if the cache already exists and based on that it would either use `bumpRecency` or just go with adding the cache using `addToCache`.  What do you think?

Reply:
Looks good! Maybe just worth a `console.warn` in the unexpected case